video/out/vulkan/ra_vk.c

```diff
@@ -689,8 +689,20 @@ struct ra_buf_vk {
     // "current" metadata, can change during course of execution
     VkPipelineStageFlags current_stage;
     VkAccessFlags current_access;
+    // Arbitrary private data for the creator of a buffer
+    void *priv;
```

Comment:
As discussed, this is not really the cleanest approach. But the alternative was copy/pasting the `ra_buf_pool` code and neither one is really preferable to the other. Either way, this code is going to be removed once we switch to a texture-based asynchronous 1-copy mapper design, pending needed changes in the nvidia drivers.